### ISSUE TITLE
Failed to resolve directive: touch

### ISSUE THREAD

**Issue by skjnldsv** [jerrybendy/vue-touch-events#30]
Using this library into another library and then importing the 2nd library doesn't work.

We have a Modal component on the https://github.com/nextcloud/nextcloud-vue library and we wanted to use your library, though when importing the modal component on another app, we get the `Failed to resolve directive: touch` error.

Though importign and using your library on the `nextcloud-vue` library is done properly:
```js
import Vue from 'vue'
import Vue2TouchEvents from 'vue2-touch-events'

Vue.use(Vue2TouchEvents)
```

I'm guessing this is because you don't export your plugin as a ssr module or something webpack can understand and pack and bundle on other libraries. But I might be wrong! :)

**Comment by jerrybendy** [comment]
Maybe I got your point, but I didn't use ES6's `export`. I'm not sure whether it works or not if I change `module.exports` to `export`.

```diff
if (typeof module === 'object') {
-    module.exports = vueTouchEvents
+   exports.install = vueTouchEvents.install
+   exports.vueTouchEvents = vueTouchEvents
}
```

**Comment by skjnldsv** [comment]
It sill behave the same :confused:

**Comment by jerrybendy** [comment]
You're right. I should restructure the code to separate the `install` function. But I can't understand the code you referenced.

```js
export const directive = {
	options: defaultOptions,
	bind,
	update: bind,
	unbind (el) {
		destroyTooltip(el)
	},
}
```

Why does `v-tooltip` export a `directive`? What does it do? 🤔

**Comment by skjnldsv** [comment]
This is the default directive that vue can use in the `directives` component.
https://vuejs.org/v2/guide/custom-directive.html

```js
props: {
  ...
},
directives: {
  focus: {
    // directive definition
    inserted: function (el) {
      el.focus()
    }
  }
}
```

**Comment by jerrybendy** [comment]
OK, I see. I will restructure my code to support it 😬

**Comment by Fragilem17** [comment]
Maybe first pull my feature.. might be easier then afterwards.

**Comment by ChristophAnastasiades** [comment]
Any news on this? I am having the same problem.

**Comment by jerrybendy** [comment]
I added a commit here #39 to support custom directives, but I can't make sure it can work well or not 😅

```
const touchDirective = require('vue2-touch-events').touchDirective

Vue.directive('touch', touchDirective)
```

**Comment by ptrckdev** [comment]
Not sure if this is a similar issue but i get the same warning so i don't want to make a new post.

Clean install of vue with the cli tool and just adding

`Vue.use(Vue2Touch)`

and adding this to a div

`v-touch:swipe="swipeHandler"`

breaks the code with

`[Vue warn]: Failed to resolve directive: touch`

What I can do, is to run the dev server without the v-touch line, wrtie it back into my div and do a hot reload, then everything works like it should.
If i reload my tab it breaks again tho.

**Comment by jerrybendy** [comment]
@ptrckdev  Which version did you use? 

```
import Vue from 'vue'
import Vue2TouchEvents from 'vue2-touch-events'

Vue.use(Vue2TouchEvents)
```

This will run `install` function and register the `v-touch` directive.

**Comment by ptrckdev** [comment]
Fixed, didnt work with Vue for a year now and put the Vue.use under new Vue() in my main.js 🙄

**Comment by jerrybendy** [comment]
@ptrckdev  Yep.  You can't put `Vue.use` under `new Vue`.

**Comment by thequiet** [comment]
> I added a commit here #39 to support custom directives, but I can't make sure it can work well or not 😅
> 
> ```
> const touchDirective = require('vue2-touch-events').touchDirective
> 
> Vue.directive('touch', touchDirective)
> ```

What aspects are you uncomfortable with? It seems with the latest version of vuetify-loader (at least in our build), the v-touch directive won't work as it conflicts with their own v-touch directive.

**Comment by aezur** [comment]
I had the same issue and it was caused by having 2  plugins in the same `Vue.use`.
I changed `Vue.use(Argon, Vue2TouchEvents);` to
```
Vue.use(Argon);
Vue.use(Vue2TouchEvents);
```

**Comment by jerrybendy** [comment]
It seems to be a big problem. I will resolve this issue ASAP

**Comment by BrianDavidYork** [comment]
Not sure if this is that same issue.  When testing with vue test utils and jest I get this error for each instance of v-touch in my html:
```
      (found in <Component>)
    console.error node_modules/vue/dist/vue.runtime.common.dev.js:621
      [Vue warn]: Failed to resolve directive: touch
```
I am importing it into my test like this:
`const touchDirective = require("vue2-touch-events").touchDirective;`  
```
describe("TEST", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);
  localVue.directive(touchDirective);

  const factory = propsData => {
    return mount(Component, {
      localVue
    });
  };

  it("isCompleted", async () => {
    const wrapper = factory();
    ...
```

Any idea how to fix this?

**Comment by jerrybendy** [comment]
@BrianDavidYork   vue2-touch-events now exports a `install` function only. So that you can do this:

```js
const vue2TouchEvents = require('vue2-touch-events');
// ...
localVue.use(vue2TouchEvents);
// ...
```

**Comment by alekbless** [comment]
Is there any option to define a custom directive when installing? I'am using Vuetify and the component "v-touch" seems to override this package.

```javascript
import Vue2TouchEvents from 'vue2-touch-events';
Vue.use(Vue2TouchEvents);
```
